hms-api/src/service/idea_comment-service.ts

```diff
@@ -0,0 +1,130 @@
+import Uuid from '../util/Uuid';
+import NotFoundError from '../error/NotFoundError';
+import {getIdea, ideaExists} from '../repository/idea-repository';
+import {getUser, userExists} from '../repository/user-repository';
+import {
+  commentAlreadyExists,
+  getComment,
+  putComment,
+  deleteComment,
+  listComments,
+} from '../repository/idea_comment-repository';
+import ValidationError from '../error/ValidationError';
+import Comment from '../repository/domain/IdeaComment';
+import ReferenceNotFoundError from '../error/ReferenceNotFoundError';
+import CommentResponse from '../rest/Comment/CommentResponse';
+import CommentDeleteResponse from '../rest/Comment/CommentDeleteResponse';
+import CommentListResponse from '../rest/Comment/CommentListResponse';
+import ideaComment from '../repository/domain/IdeaComment';
+
+export async function createComment(
+  ideaId: Uuid,
+  userId: Uuid,
+  text: string,
+  replyTo: Uuid,
+): Promise<Comment> {
+  if (!(await ideaExists(ideaId))) {
+    throw new ReferenceNotFoundError(`Idea with id: ${ideaId} not found`);
+  } else if (
+    replyTo &&
+    !(await commentAlreadyExists(new Comment(userId, ideaId, text, replyTo)))
+  ) {
+    throw new ReferenceNotFoundError(`Comment with id: ${replyTo} not found`);
+  } else if (!(await userExists(userId))) {
+    throw new ReferenceNotFoundError(`User with id: ${userId} not found`);
+  }
+
+  const comment = new Comment(userId, ideaId, text, replyTo);
+  const result = comment.validate();
+  if (result.hasFailed()) {
+    throw new ValidationError('Cannot create Comment', result);
+  }
+
+  await putComment(comment);
+
+  return comment;
+}
+
+export async function getCommentResponse(id: Uuid): Promise<CommentResponse> {
+  const comment = await getComment(id);
+
+  let user;
+  try {
+    user = await getUser(comment.userId);
+  } catch (e) {
+    throw new ReferenceNotFoundError(
+      `Cannot get Comment with id ${id}, ` +
+        `unable to get User with id ${comment.userId}`,
+    );
+  }
+
+  let idea;
+  try {
+    idea = await getIdea(comment.ideaId);
+  } catch (e) {
+    throw new ReferenceNotFoundError(
+      `Cannot get Comment with id ${id}, ` +
+        `unable to get Idea with id ${comment.ideaId}`,
+    );
+  }
+
+  return CommentResponse.from(comment, user);
+}
+
+export async function getCommentListResponse(
+  ideaId: Uuid,
+): Promise<CommentListResponse> {
+  if (!(await ideaExists(ideaId))) {
+    throw new NotFoundError(`Idea with id: ${ideaId} not found`);
+  }
+
+  const comments = await listComments(ideaId);
+
+  //Todo: evaluate how we handle comments with deleted users
+  let users;
+  try {
+    users = await Promise.all(
+      comments.map((comment) => getUser(comment.userId)),
+    );
+  } catch (e) {
+    throw new ReferenceNotFoundError(
+      `Cannot get Comment list, ` +
+        `unable to get User with id ${users.userId}`,
```

Comment:
`users` would be a list with many User objects in it.  And if this were to catch it will still be undefined.  You probably should print the list of ALL users in the comments (`comments.